bokeh/core/_templates/autoload_nb_js.js

```diff
@@ -1,5 +1,116 @@
 {% extends "autoload_js.js" %}
 
+{% block register_mimetype %}
+
+  var MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';
+  var CLASS_NAME = 'output_bokeh rendered_html';
+
+  /**
+   * Render data to the DOM node
+   */
+  function render(props, node) {
+    if (props.data['div'] !== undefined) {
+      var div = document.createElement("div");
+      div.innerHTML = props.data["div"];
+      node.appendChild(div);
+    }
+
+    if (props.data['script'] !== undefined) {
+      var script = document.createElement("script");
+      script.textContent = props.data["script"];
+      node.appendChild(script);
+    }
+  }
+
+  /**
+   * Handle when an output is cleared or removed
+   */
+  function handleClearOutput(event, handle) {
+    var cell = handle.cell;
+
+    var id = cell.output_area._bokeh_element_id;
+    var server_id = cell.output_area._server_id;
+    // Clean up Bokeh references
+    if (id !== undefined) {
+      Bokeh.index[id].remove();
+      delete Bokeh.index[id];
+    }
+
+    if (server_id !== undefined) {
+      // Clean up Bokeh references
+      var cmd = "from bokeh.io import _state; print(_state.uuid_to_server['" + server_id + "'].get_sessions()[0].document.roots[0]._id)";
```

Comment:
I don't think there will ever normally be more than one session. In order for there to be multiple sessions a user would have to jump through many hoops, enough that I'd say they are on their on at that point. I am less certain that there would be only one root ever. Certainly that's the current common case but once we make it possible to template roots into different parts of the dom separately I can imagine folks wanting more sophisticated embedded apps